tensorboard/components/tensor_widget/demo/demo.html

```diff
@@ -40,8 +45,72 @@ <h1>TensorWidget Demo</h1>
         'org_tensorflow_tensorboard/tensorboard/components/tensor_widget/tensor-widget';
 
       require([TENSOR_WIDGET_MODULE_PATH], function(tensorWidgetLib) {
```

Comment:
Where can I learn more about this syntax for 'require'?  Is it something provided by the bundler?

Reply:
Yea, that was the part that concerned me a bit but since it was a demo, I let it slide.
It is often provided by webpack or browserify or others. It is quite common for bundlers to provide an implementation of `require`. But IIRC, unless you have explicitly tell some bundlers that you will have multiple bundles or use code splitting, they would not expose the require method on the `window` (the behavior wrt multiple bundle and `window.require` is quite fascinating, too :) ). Since, in this case, it is magically provided, I thought it was okay.

Reply:
@bileschi The `require-define` syntax is the JavaScript way of defining and loading modules before ES Modules became widely supported and TypeScript became a thing. It's called AMD (Asynchronous module definition). We are talking about "prehistoric" time of 2014 or so.

See https://addyosmani.com/writing-modular-js/ for a good tutorial.